core/src/Kokkos_Macros.hpp

```diff
@@ -171,9 +171,14 @@
 
 #define KOKKOS_IMPL_HIP_CLANG_WORKAROUND
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wc99-designator"
```

Comment:
We need to discuss this.

Reply:
I can live with this since its just around the include of the hip system header files isn't it?

Reply:
I don't see this warning though

Reply:
I want to see it and tell AMD to fix it.  Also we should not be seeing it if `hipcc` is configured properly and uses `-isystem`

Reply:
I am getting
```bash
cd /ccs/home/darndt/kokkos/build/core/src && /opt/rocm/bin/hipcc   -I/ccs/home/darndt/kokkos/build/core/src -I/ccs/home/darndt/kokkos/core/src -I/ccs/home/darndt/kokkos/build  -Werror   -o CMakeFiles/kokkoscore.dir/impl/Kokkos_Core.cpp.o -c /ccs/home/darndt/kokkos/core/src/impl/Kokkos_Core.cpp
In file included from /ccs/home/darndt/kokkos/core/src/impl/Kokkos_Core.cpp:44:
In file included from /ccs/home/darndt/kokkos/core/src/Kokkos_Core.hpp:50:
In file included from /ccs/home/darndt/kokkos/core/src/Kokkos_Core_fwd.hpp:51:
In file included from /ccs/home/darndt/kokkos/core/src/Kokkos_Macros.hpp:174:
In file included from /opt/rocm/include/hip/hip_runtime.h:56:
In file included from /opt/rocm/include/hip/hcc_detail/hip_runtime.h:106:
/opt/rocm/include/hip/hcc_detail/texture_functions.h:173:5: error: array designators are a C99 extension [-Werror,-Wc99-designator]
    [HIP_AD_FORMAT_NOT_INITIALIZED] = 1  ,
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
where
```
$ hipcc --version
HIP version: 3.0.19493-36529b16
HCC clang version 10.0.0 (/data/jenkins-workspace/centos_pipeline_job_3.0/rocm-rel-3.0/rocm-3.0-6-20191216/centos/external/hcc-tot/llvm-project/clang 40756364719e83a2ddd5abe0affe700723cdd852) (based on HCC 3.0.19493-75ea952e-40756364719e )
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/rocm/hcc/bin
```

Reply:
This has been fixed a couple of months ago https://github.com/ROCm-Developer-Tools/HIP/pull/1694

Reply:
This fix looks like it will be present in the 3.1 release  
-Nick